src/App.js

```diff
@@ -1,25 +1,80 @@
-import logo from './logo.svg';
-import './App.css';
+import logo from "./logo.svg";
+import "./App.css";
+import React, { useState } from "react";
+import CoinImage from "./components/CoinImage";
+import WinMessage from "./components/WinMessage";
 
-function App() {
+// function App() {
+//   return (
+//     <div className="App">
+//       <header className="App-header">
+//         <img src={logo} className="App-logo" alt="logo" />
+//         <p>
+//           Edit <code>src/App.js</code> and save to reload.
+//         </p>
+//         <a
+//           className="App-link"
+//           href="https://reactjs.org"
+//           target="_blank"
+//           rel="noopener noreferrer"
+//         >
+//           Learn React
+//         </a>
+//       </header>
+//     </div>
+//   );
+// }
+
+// export default App;
+
+export default function App() {
+  const [heads, setHeads] = useState(0);
+  const [tails, setTails] = useState(0);
+  const [coin, setCoin] = useState(null); // coin is the value associated w H or T coin // setCoin is the function that changes the value of coin
+
+  function Reset() {
+    setTails(0);
+    setHeads(0);
+    setCoin(null);
+  }
+
+  function flipCoin() {
+    // assign random value where 0 = heads & 1 = tails
+    if (heads === 5 || tails === 5) {
+      return Reset();
+    }
+    var flip = Math.floor(Math.random() * 2);
+    console.log(flip);
+    setCoin(flip); // now the value of coin will be 0 or 1 bc we passed
+    if (flip === 0) {
+      return setHeads(heads + 1);
```

Comment:
🔴 There's a subtle bug here due to a combination of how JS and React work. Any state variable could potentially be out of date. In this case, `heads` might not have been updated by the time the user clicks the button again, resulting in an inaccurate count. To work around this, `setState` accepts a function and provides the latest value:

```suggestion
      return setHeads(value => value + 1);
```

Here's an article that explains this more in depth: https://beta.reactjs.org/reference/react/useState#updating-state-based-on-the-previous-state